### PR TITLE
docs: Update roadmap

### PR DESCRIPTION
**Pull request by max-sixty**
This is now ready for review. I'm happy to discuss things here. It's also fine if we merge without complete agreement and discuss on the next dev call.

### CODE REVIEW

**Comment by max-sixty** [comment]
Rendered: 
image

**Comment by aljazerzen** [comment]
I've pushed two commits to roadmap branch - i don't have access to this one.

There is still quite a few things to do in language design, but I think it'll be fastest to discuss that on the call.

**Comment by max-sixty** [comment]
> I've pushed two commits to roadmap branch - i don't have access to this one.

Hmmm, this is ticked, so I would have thought you had access...

image

I've pulled it in though

(as you can see I've started using a fork rather than the main repo the past few weeks)

**Comment by aljazerzen** [comment]
Oh, I didn't know I had write permissions, it does work.

> (as you can see I've started using a fork rather than the main repo the past few weeks)

Why though?

**Comment by max-sixty** [comment]
> Why though?

There was no downside to using a fork + I didn't want to be "special" just because I had been using the main repo when it was under @max-sixty in early 2022. 

And I wanted to ensure I wasn't missing anything with the contributor experience; e.g. I only recently realized that folks were getting failures on GitHub Pages deploys on their own fork (I fixed that by turning it off in forks)

If you prefer using the main repo, totally fine ofc!

File: website/content/roadmap.md [hunk 1]
```diff
@@ -15,77 +15,133 @@ url: roadmap
...
+representation) and then use that to apply the transformations to an in-memory
+dataframe of a performance-optimized library (such as
+[Polars](https://www.pola.rs/)) or a Google Sheets spreadsheet. Alternatively,
+we could even convert RQ to [Substrait](https://substrait.io/).
 
-`pyprql` is a step into direction of a general data handling program. Building
-on this, we want to build a tool that can read many data sources, offers syntax
-highlighting, auto-complete and type-inference using information from database's
-schema.
+#### PRQL as a tool
```
Comment (aljazerzen) [inline website/content/roadmap.md:136]: @snth do you want to mention prql-query? I would but I know you've been suggesting we delay

File: website/content/roadmap.md [hunk 2]
```diff
@@ -15,77 +15,133 @@ url: roadmap
...
+are planning
+[a few major features](https://github.com/PRQL/prql/issues?q=is%3Aopen+is%3Aissue+label%3Amajor-feature+label%3Alanguage-design)
+that will give PRQL the feeling of a real programming language and elevate it in
+[the chomsky hierarchy](https://en.wikipedia.org/wiki/Chomsky_hierarchy).
+Honorable mentions here are recursive CTEs (or rather functions), algebraic type
+system, pre-specified join conditions and regex.
+
+Note that these features will probably inflict breaking changes with each minor
+release before we stabilize the 1.0, the first indefinitely supported language
+edition.
```
Comment (aljazerzen) [inline website/content/roadmap.md:118]: @max-sixty I've removed "Rethinking joins". I feel that this is just one of the language features.

**Review by aljazerzen — APPROVED** [review]
@max-sixty LGTM, feel free to merge

File: website/content/roadmap.md [hunk 3]
```diff
@@ -151,86 +153,3 @@ means putting some things out of scope:
...
-features.
-
-There are still some features that are missing: a native regex operator,
-
-On this foundation we are planning to build advanced features like type
-checking, function currying,  -->
-
-<!--
-
-TODO: are we planning to offer these? Where the underlying DB doesn't offer them, it'll be quite hard!
```
Comment (max-sixty) [inline website/content/roadmap.md]: @snth maybe you added these? Feel free to add them back / open a discussion...

File: website/content/roadmap.md (hunk 2 above)
Comment (max-sixty) [reply website/content/roadmap.md:118]: OK, agree

**Pull request merged**